src/panel/applets/status/PowerIndicator.vala

```diff
@@ -196,12 +315,46 @@ public class PowerIndicator : Gtk.Bin {
 
 		client = new Up.Client();
 
+		Bus.watch_name(BusType.SYSTEM, POWER_PROFILES_DBUS_NAME, BusNameWatcherFlags.NONE, has_power_profiles, lost_power_profiles);
+
 		this.sync_devices();
 		client.device_added.connect(this.on_device_added);
 		client.device_removed.connect(this.on_device_removed);
 		toggle_show();
 	}
 
+	void has_power_profiles() {
+		if (profiles_proxy == null) {
+			Bus.get_proxy.begin<net.hadess.PowerProfilesDBus>(BusType.SYSTEM, POWER_PROFILES_DBUS_NAME, POWER_PROFILES_DBUS_OBJECT_NAME, 0, null, on_proxy_get);
+		} else {
+			// already have a proxy object
+			create_power_profiles_options();
+		}
```

Comment:
I'm not sure that this will work if the connection was lost; the proxy object might be invalid. That's why I was thinking that the widget would store the proxy object so that this class doesn't have to.

Reply:
tested- it does actually work